src/application.coffee

```diff
@@ -644,7 +653,28 @@ application.update = update = (force) ->
 			# Set the updating as finished in its own block, so it never has to worry about other code stopping this.
 			updateStatus.state = UPDATE_IDLE
 
+listenToEvents = ->
+	docker.getEventsAsync()
+	.then (stream) ->
```

Comment:
You probably need to handle errors on this stream, to stop the process potentially dying